partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/TypeEnv.kt

```diff
@@ -22,10 +23,16 @@ import org.partiql.value.stringValue
  * @property outer refers to the outer variable scopes that we have access to.
  */
 internal data class TypeEnv(
+    private val globals: Env,
```

Comment:
I think we should consider a "scope" class — [I experimented with it here.](https://github.com/partiql/partiql-lang-kotlin/tree/v1-planner/partiql-planner/src/main/kotlin/org/partiql/planner/intern/validate)

The "scope" would replace TypeEnv since TypeEnv is explicitly the name for locals rather than here we have globals+locals which makes sense.

Scope/SqlScope in this example also operates on its own internal classes rather than over stateless IR